src/clients/inst_store_upgrade_software.rb

```diff
@@ -33,10 +33,11 @@ def main
             Ops.get_string(p, "name", "")
           )
         end
+
         nil
       end
       @patterns_to_install = Builtins.filter(@patterns_to_install) do |p|
-        p != nil
+        !p.nil?
```

Comment:
Perhaps we could just use `#compact`.

Reply:
I need to check whole code as it can crash if @patterns_to_install will be nil, which is not case here as it use old filter which handle also nil.